### PR TITLE
Crc 2327

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/common/BasicGoalConfig.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.raml.maven.common;
+
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+POJO to hold config for maven plugin goal.
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/common/BasicGoalConfig.java]: Missing asterix? weird comment style...

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/generator/GenerateMojo.java [hunk 2]
```diff
@@ -1,52 +1,33 @@
...
 import java.io.File;
-import java.util.List;
 import java.util.Map;
 
 import static org.apache.maven.plugins.annotations.ResolutionScope.COMPILE_PLUS_RUNTIME;
 
-@Mojo(name = "generate", requiresProject = true, threadSafe = false, requiresDependencyResolution = COMPILE_PLUS_RUNTIME, defaultPhase = LifecyclePhase.GENERATE_SOURCES)
-public class GenerateMojo extends AbstractMojo {
+@Mojo(name = "generate", requiresDependencyResolution = COMPILE_PLUS_RUNTIME, defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class GenerateMojo extends uk.gov.justice.raml.maven.common.BasicMojo {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/generator/GenerateMojo.java]: Do we need the fully qualified import for BasicMojo?

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+@Mojo(name = "check-syntax")
+public class RamlSyntaxCheckMojo extends uk.gov.justice.raml.maven.common.BasicMojo {
+
+    RamlSyntaxValidator ramlSyntaxValidator = new RamlSyntaxValidator(new FileTreeScannerFactory());
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: Make final? - should this be private or is it accessible for testing?

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import uk.gov.justice.raml.io.FileTreeScannerFactory;
+import uk.gov.justice.raml.maven.common.BasicGoalConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+@Mojo(name = "check-syntax")
+public class RamlSyntaxCheckMojo extends uk.gov.justice.raml.maven.common.BasicMojo {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: Fully qualified name as above...

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        configureDefaultFileIncludes();
+
+        report(ramlValidationResults());
+    }
+
+    private void report(final Map<Path, ValidationResult> results) throws MojoFailureException {
+        if (results.size() == 0) {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: Should we be using isEmpty() here?

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        try {
+            return ramlSyntaxValidator.validateRamls(configuration());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Files not loaded", e);
+        }
+    }
+
+    private String printResults(Map<Path, ValidationResult> results, File sourceDirectory) {
+
+        StringBuilder sb = new StringBuilder();
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: final StringBuilder

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+    private Map<Path, ValidationResult> ramlValidationResults() throws MojoExecutionException {
+        try {
+            return ramlSyntaxValidator.validateRamls(configuration());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Files not loaded", e);
+        }
+    }
+
+    private String printResults(Map<Path, ValidationResult> results, File sourceDirectory) {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: final method parameters

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java (hunk 3 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: Accessible for testing

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java (hunk 4 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: Will fix this

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java (hunk 5 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: Will fix

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/common/BasicGoalConfig.java (hunk 1 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/common/BasicGoalConfig.java]: done

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/generator/GenerateMojo.java (hunk 2 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/generator/GenerateMojo.java]: fixed

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java (hunk 6 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: fixed

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java (hunk 7 above)
Comment (skiddykong) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxCheckMojo.java]: fixed

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    private final FileTreeScannerFactory scannerFactory;
+
+    public RamlSyntaxValidator(final FileTreeScannerFactory scannerFactory) {
+
+        this.scannerFactory = scannerFactory;
+
+    }
+
+    public Map<Path, ValidationResult> validateRamls (BasicGoalConfig config) throws IOException {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java]: final config param.

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            for(ValidationResult result : tmpResults) {
+                if(result != null) {
+                    results.put(path, result);
+                }
+            }
+        }
+        return results;
+    }
+
+    private List<ValidationResult> validationResults(Path path, BasicGoalConfig config) {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java]: final method params.

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        final Collection<Path> paths = scannerFactory.create().find(config.getSourceDirectory(), includes, excludes);
+
+        Map<Path, ValidationResult> results = new HashMap<>();
+
+        for(Path path : paths) {
+
+            List<ValidationResult> tmpResults = validationResults(path, config);
+
+            for(ValidationResult result : tmpResults) {
+                if(result != null) {
```
Comment (matt-rich) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java]: Are you adding null results back into your List above? Do we need this check, or should we stop adding null into the list if we are.

File: raml-maven-plugin/src/test/resources/validator/bad-schema/pom.xml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>raml-maven-plugin</artifactId>
+                <groupId>uk.gov.justice.maven</groupId>
+                <configuration>
+                    <sourceDirectory>src/raml</sourceDirectory>
+                    <!--<includes>-->
```
Comment (matt-rich) [inline raml-maven-plugin/src/test/resources/validator/bad-schema/pom.xml]: Do we want to remove the commented out configuration?

File: raml-maven-plugin/src/test/resources/validator/good-raml/pom.xml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>raml-maven-plugin</artifactId>
+                <groupId>uk.gov.justice.maven</groupId>
+                <configuration>
+                    <sourceDirectory>src/raml</sourceDirectory>
+                    <!--<includes>-->
```
Comment (matt-rich) [inline raml-maven-plugin/src/test/resources/validator/good-raml/pom.xml]: Commented out config.

File: raml-maven-plugin/src/test/resources/validator/multiple-ramls/pom.xml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>raml-maven-plugin</artifactId>
+                <groupId>uk.gov.justice.maven</groupId>
+                <configuration>
+                    <sourceDirectory>src/raml</sourceDirectory>
+                    <!--<includes>-->
```
Comment (matt-rich) [inline raml-maven-plugin/src/test/resources/validator/multiple-ramls/pom.xml]: Commented out code.

File: raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>raml-maven-plugin</artifactId>
+                <groupId>uk.gov.justice.maven</groupId>
+                <configuration>
+                    <sourceDirectory>src/raml</sourceDirectory>
+                    <!--<includes>-->
```
Comment (matt-rich) [inline raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml]: Commented out code

File: raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+    <artifactId>project-to-test</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>raml-maven-plugin</artifactId>
+                <groupId>uk.gov.justice.maven</groupId>
+                <configuration>
+                    <sourceDirectory>src/raml</sourceDirectory>
```
Comment (matt-rich) [inline raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml]: Is it worth making src/raml a pom property - in the parent ? - If you agree, this also applied to the other pom files above.

File: raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml (hunk 14 above)
Comment (jaceko) [inline raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml]: fixed

File: raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml (hunk 15 above)
Comment (jaceko) [inline raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml]: I don't think it is worth here. We are not using parent pom in test pom files.

File: raml-maven-plugin/src/test/resources/validator/multiple-ramls/pom.xml (hunk 13 above)
Comment (jaceko) [inline raml-maven-plugin/src/test/resources/validator/multiple-ramls/pom.xml]: fixed

File: raml-maven-plugin/src/test/resources/validator/good-raml/pom.xml (hunk 12 above)
Comment (jaceko) [inline raml-maven-plugin/src/test/resources/validator/good-raml/pom.xml]: fixed

File: raml-maven-plugin/src/test/resources/validator/bad-schema/pom.xml (hunk 11 above)
Comment (jaceko) [inline raml-maven-plugin/src/test/resources/validator/bad-schema/pom.xml]: fixed

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java (hunk 10 above)
Comment (jaceko) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java]: fixed

File: raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java (hunk 8 above)
Comment (jaceko) [inline raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/validator/RamlSyntaxValidator.java]: fixed

File: raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml (hunk 15 above)
Comment (matt-rich) [inline raml-maven-plugin/src/test/resources/validator/no-raml/pom.xml]: Fair enough.

**Pull request merged**